packages/@uppy/transloadit/src/Assembly.js

```diff
@@ -62,6 +65,59 @@ class TransloaditAssembly extends Emitter {
     this.close()
   }
 
+  #connectServerEvent () {
+    this.#sse = new EventSource(this.status.assembly_ssl_url)
+
+    this.#sse.addEventListener('open', () => {
+      if (this.socket) {
+        this.socket.disconnect()
+        this.socket = null
+      }
+      clearInterval(this.pollInterval)
+      this.pollInterval = null
+    })
+
+    /*
+     * The event "message" is a special case, as it
+     * will capture events without an event field
+     * as well as events that have the specific type
+     * other event type.
+     */
+    this.#sse.addEventListener('message', (e) => {
+      if (e.data === 'assembly_finished') {
+        this.#onFinished()
+      }
+
+      if (e.data.startsWith('assembly_upload_finished:')) {
+        const file = JSON.parse(e.data.slice('assembly_upload_finished:'.length))
```

Comment:
try/catch `JSON.parse` in case of an error?

Reply:
Not sure it'd be useful, that would only happen if there's a bug on Transloadit side, at which point throwing an error seems like the correct behavior.